src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -1526,12 +1529,12 @@ public boolean hasEnableSecurityForm() {
 
         @RequirePOST
         public FormValidation doValidate(StaplerRequest req) throws Exception {
-            if (!Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER)) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
                 // require admin to test
                 return FormValidation.ok();
             }
             // extract the submitted details
-            JSONObject json = JSONObject.fromObject(IOUtils.toString(req.getInputStream()));
+            JSONObject json = JSONObject.fromObject(IOUtils.toString(req.getInputStream(), Charset.defaultCharset()));
```

Comment:
I realize this was just preserving existing behavior, so specific action item here, but I suspect the original behavior was wrong. In the few cases in core where we use `IUtils#toString` with a `StaplerRequset` object, we always pass in either `req.getCharacterEncoding()` or UTF-8 as the encoding.

Reply:
Is it safe to assume UTF-8 or is `getCharacterEncoding()` the better choice?

Reply:
I think `ServletRequest#getCharacterEncoding` is technically the more correct option since the HTTP request could come in via any encoding in theory, but I think the HTTP requests are usually (always?) UTF-8 in practice.

Reply:
Thanks! This a good chance for improvement; I'm going to submit a PR.

Reply:
--> #174